sdks/python/apache_beam/ml/inference/model.py

```diff
@@ -0,0 +1,74 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+
+VALID_PYTORCH_DEVICE_TYPES = ['CPU', 'GPU']
+VALID_SKLEARN_SERIALIZATION_TYPES = ['JOBLIB', 'PICKLE']
+
+
+class RunInferenceModel(object):
+  '''
+  Base class for the RunInference Model.
+  This class contains information about
+  a model that is needed
+  '''
+  def __init__(self, model_url):
+    self._model_url = model_url
+    self._validate_model()
+
+  @abc.abstractmethod
+  def _validate_model(self):
+    raise NotImplementedError("Please implement _validate_model")
+
+
+class PytorchModel(RunInferenceModel):
```

Comment:
It feels like PytorchModel and SklearnModel should be in their own file, only imported if one or the other is needed, rather than having a single module that holds all model types.

Reply:
Agreed. I'll refactor.

Reply:
Let's keep comments like this open until they're done for easier tracking.

Reply:
These implementations will be done in separate PRs ([pytorch](https://issues.apache.org/jira/browse/BEAM-13984), [sklearn](https://issues.apache.org/jira/browse/BEAM-13983)) after this PR, as well as the PR for the base class #16970 are merged.